Geometry.cs

```diff
@@ -144,14 +158,46 @@ class Drawing {
 
    public Bound2 Bound {
       get {
-         if (mBound.IsEmpty) mBound = new (Polys.Select (a => a.Bound));
+         if (mBound.IsEmpty) mBound = new (ConvexHull);
          return mBound;
       }
    }
    Bound2 mBound;
 
-   public Bound2 GetBound (Matrix2 xfm) 
-      => new Bound2 (Polys.SelectMany (a => a.Pts.Select (p => p * xfm)));
+   IReadOnlyList<Point2> ConvexHull { 
+      get {
+         if (!mConvexHull.Any ()) ComputeConvexHull (Polys.SelectMany (a => a.Pts));
+         return mConvexHull;
+      } 
+   }
+   List<Point2> mConvexHull = new ();
+
+   void ComputeConvexHull (IEnumerable<Point2> pts) {
+      Point2 mPtY = pts.Min ();
+      List<Point2> mCH = new ();
+      List<Point2> tpts = pts.OrderBy (mPtY.AngleTo).ToList ();
+      mCH.Add (mPtY);
+      foreach (var p in tpts) {
+         mCH.Add (p);
+         CheckLastAngle ();
+      }
+
+      void CheckLastAngle () {
+         int count = mCH.Count;
+         while (count > 3) {
+            Vector2 v1 = new (mCH[count - 1].X - mCH[count - 2].X, mCH[count - 1].Y - mCH[mCH.Count - 2].Y);
+            Vector2 v2 = new (mCH[count - 2].X - mCH[count - 3].X, mCH[count - 2].Y - mCH[mCH.Count - 3].Y);

```

Comment:
Use mCH[^1], mCH[^2] where possible to simplify this.

Reply:
Changed